lsst_efd_client/efd_helper.py

```diff
@@ -377,9 +418,67 @@ async def select_packed_time_series(self, topic_name, base_fields, start, end,
         result = await self.select_time_series(topic_name, field_list+[ref_timestamp_col, ],
                                                start, end, is_window=is_window, index=index)
         vals = {}
-        for field in base_fields:
-            df = merge_packed_time_series(result, field, ref_timestamp_col=ref_timestamp_col,
+        for f in base_fields:
+            df = merge_packed_time_series(result, f, ref_timestamp_col=ref_timestamp_col,
                                           internal_time_scale=self.internal_scale)
-            vals[field] = df[field]
+            vals[f] = df[f]
         vals.update({'times': df['times']})
         return pd.DataFrame(vals, index=df.index)
+
+    async def get_schema_topics(self):
+        """Return a list of topics in the SchemaRegistry
+
+        Returns
+        -------
+        result : `list` of `str`
+            A `list` of the names of all SAL topics in the registry
+        """
+        async with aiohttp.ClientSession() as http_session:
+            registry_api = RegistryApi(
+                session=http_session, url=self.schema_registry
+            )
+            subjects = await registry_api.get('subjects')
+            ret = []
+            for s in subjects:
+                if not s.startswith('lsst.sal'):  # Only pass on sal topics
+                    continue
+                ret.append(s[:-6])  # strip off the -value part so these are useful as is
+            return ret
+
+    async def get_schema_dict(self, topic):
+        """Givent a topic, get a list of dictionaries describing the fields
+
+        Parameters
+        ----------
+        topic : `str`
+            The name of the topic to query.  A full list of valid topic names
+            can be obtained using ``get_schema_topics``.
+
+        Returns
+        -------
+        result : `list` of `dict`
+            A `list` of dictionaries holding the schema information for this topic
+        """
+        async with aiohttp.ClientSession() as http_session:
+            registry_api = RegistryApi(
+                session=http_session, url=self.schema_registry
+            )
+            schema = await registry_api.get_schema_by_subject(f'{topic}-value')
+            return schema['schema']['__named_schemas'][topic]['fields']
+
+    async def get_schema_fields(self, topic):
```

Comment:
I would suggest `EFDClient.get_topic_schema(topic_name)`  or just `EFDClient.get_schema(topic_name)`  for the method name.  This would retrieve the list of fields, field descriptions and units from the Schema Registry for a given topic. 

This could also be returned as a Pandas Dataframe, so that it would be easy to get a single field description and units from there, and would be nicely rendered in the notebook.

Reply:
OK.  I like the idea of returning a pandas dataframe

Reply:
OK.  I've implemented this as `get_schema` and am now returning a dataframe.